sdk/keyvault/keyvault/inc/az_keyvault.h

```diff
@@ -5,43 +5,103 @@
 #define AZ_KEYVAULT_H
 
 #include <az_contract.h>
+#include <az_credential.h>
+#include <az_http_pipeline.h>
+#include <az_http_request_builder.h>
 #include <az_result.h>
 #include <az_span.h>
+#include <az_str.h>
 
 #include <stdlib.h>
 
 #include <_az_cfg_prefix.h>
 
-typedef struct {
-  az_span uri;
-  // TODO: Azure Credentials
-} az_keyvault_keys_client;
+extern az_span const AZ_KEY_VAULT_NONE_TYPE;
+extern az_span const AZ_KEY_VAULT_KEY_TYPE;
+extern az_span const AZ_KEY_VAULT_SECRET_TYPE;
+extern az_span const AZ_KEY_VAULT_CERTIFICATE_TYPE;
+extern az_span const AZ_KEY_VAULT_URL_PATH_SEPARATOR;
+typedef enum {
```

Comment:
Do these map to values sent to the service? Or, do they get converted to strings that get sent to the service?
Whatever happens should be documented and we probably need to discuss how to handle client/server enum types because they usually have to be forward compatible.